File: Week - 1/Assignment - 3/Bubble_Sort.py
def bubble_sort(arr):
    swaps = 0
    n = len(arr)

    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swaps += 1

    return swaps


# Input
n = int(input("Enter the number of elements: "))

arr = []

print("Enter the elements:")

for i in range(n):
    element = int(input(f"Element {i + 1}: "))
    arr.append(element)

# Sort the array
swap_count = bubble_sort(arr)

# Output
print("Sorted array:", arr)
print("Number of swaps:", swap_count)

File: Week - 1/Assignment - 3/BubbleSort.cpp
#include <iostream>
using namespace std;

int bubbleSort(int arr[], int n) {
    int swaps = 0;

    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {

            if (arr[j] > arr[j + 1]) {
                // Swap adjacent elements
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;

                swaps++;
            }
        }
    }

    return swaps;
}

int main() {
    int n;

    cout << "Enter the number of elements: ";
    cin >> n;

    int arr[n];

    cout << "Enter the elements:\n";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    int swapCount = bubbleSort(arr, n);

    cout << "Sorted array: ";

    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }

    cout << "\nNumber of swaps: " << swapCount << endl;

    return 0;
}